### DISCUSSION TITLE
Transporter 120 and 1024 pxl error help!

### DISCUSSION THREAD

**Discussion by ZhorovEgor** [airsdk/Adobe-Runtime-Support#3199] [General]
Howdy fellas! 
After launching the project in Adobe Animate AIR from Harmon and deploying the project in TRANSPORTER, I got this error saying that the project lacks images of 120 and 1024 pixels in size, although I specified these sizes in the icons section (screenshots below), help! 

I decided to open the project (rename the IPA to ZIP and open it simply as an archive) to see if I have pictures, it turned out that when forming the pictures are not added, but if we bring the IPA to ZIP, then we will not be able to send the project back for verification (integrity will be violated)

I will also leave the assets in text form below:

The resource could not be verified (90022)
The required icon file is missing. The package does not include an application icon for iPhone / iPod Touch with a size of exactly 120x120 pixels in .png format for iOS versions >= 10.0. To support older versions of iOS, the icon may be required in a package that is not included in the resource catalog. Make sure that the Info.plist file contains the appropriate entries referring to this file. You see https://developer.apple.com/documentation/bundleresources/information_property_list/user_interface (CODE: e9a3cb0c-050F-439d-b254-2520f1add472)

Asset Verification Error (90704)
The app icon is missing. Add a large 1024 by 1024 pixel PNG application icon to the catalog of applications created for iOS, iPadOS, or watchOS. Without this icon, applications cannot be submitted for verification. Details on the website: https://developer.apple.com/documentation/xcode/managing-assets-with-asset-catalogs . (ID: 195275c1-3a9e-4167-b2f5-4157c500c399)

![photo_2024-04-23_19-53-08 (2)](https://github.com/airsdk/Adobe-Runtime-Support/assets/63095086/52439be8-117d-4cc7-9440-43c75bd79595)
![photo_2024-04-23_19-53-08](https://github.com/airsdk/Adobe-Runtime-Support/assets/63095086/515a12cf-a824-44bd-ac3f-9f8d7a5fca3d)
![photo_2024-04-23_19-53-07](https://github.com/airsdk/Adobe-Runtime-Support/assets/63095086/eee7de87-852a-4590-a340-0be40efe0aea)

**Comment by FliplineStudios** [comment]
Someone mentioned a similar problem and a solution here:
https://github.com/airsdk/Adobe-Runtime-Support/discussions/3162#discussioncomment-9073344

Are you using Xcode 14 or 15 to create your Assets.car file where you just added one "Single Size" App Icon instead of the usual "All Sizes" where you add a bunch of separate images?  Sounds like there can sometimes be an issue with app validation if you're using that and your iOS target is less than 12.0 (it's also mentioned in the Xcode 14 Release Notes "Known Issues", it doesn't seem to have been fixed in newer releases).

**Reply by ZhorovEgor** [reply]
Good afternoon! I didn't use Xcode in principle, what is it for if I do everything through Adobe Animate? Tell us about Assets.car where to get it (under what conditions does it appear)?

**Reply by FliplineStudios** [reply]
All iOS apps need an Assets.car file to include your app icons (and assets for your LaunchScreen storyboard).  This is separate from defining all of the individual app icons within Animate or application descriptor files (though those are also required).  It has been a requirement since iOS 11 I believe, and usually you would use Xcode to create this file to then be packaged along with your app.  There are also some other tools and scripts to generate one if you don't use Xcode for it, but I don't believe Animate or the AIR SDK will create the Assets.car file themselves.

Distriqt has some more info here (I use "Method 2" myself):
https://docs.airnativeextensions.com/docs/tutorials/ios-icons-assets-car/

**Reply by ZhorovEgor** [reply]
As I understood from the description, assets.car should be in the application directory. But I can't interact with the structure directly, because I get a whole compiled IPA file at the output. Having assets.car ready on hand, how can I cram it into the application?

**Reply by ZhorovEgor** [reply]
I can repeat someName.ipa in someName.zip and open as archive, but TRANSPORTER not access refreshed file.

**Reply by FliplineStudios** [reply]
Within Animate there should be a place where you add other files to be included and packaged within the app, I believe it's in the AIR for iOS Settings in the "General" section and labeled "Included Files" or something similar, where you'll also see your swf and app xml already listed.  Your Assets.car file that you create would have to be included there to be packaged into the ipa.

**Reply by ZhorovEgor** [reply]
You right! Animate have some place to add other files! I just include on this space my new Assets.car file which i create on https://www.kxapp.com/appicon.html
